Adds tacticool helmet lights, adds siren helmet as premium item

9600bauds opened the pull request:
You may now attach flashlights to security helmets, the HoS' helmet, the Warden's helmet, and SWAT helmets. To unattach the flashlight, use a screwdriver.
New flashlight attachment sprites courtesy of Blithering (see #9279) (closes #9279) (closes #9462).
Added siren helmets as a premium item from Sec vending machines. When used, they WOOP WOOP due to lack of better sound: 0:00 in https://www.youtube.com/watch?v=_BxBs4f4RIU

I had to make Sec helmets into a subtype, did so in such a way that other helmets could easily get new attachments if sprites are provided in the future. Modifies the maps to accomplish this

92% tested

Pictures:
![helmet light](https://cloud.githubusercontent.com/assets/15067109/14321856/cf08d5d4-fc12-11e5-9f74-8ce2d3a044ae.png)
![mspaint_2016-04-12_01-58-16](https://cloud.githubusercontent.com/assets/6526157/14450184/1a78b58c-0052-11e6-833c-64fb6d4ad66c.png)

9600bauds commented on code/modules/clothing/head/misc_special.dm:
```diff
@@ -5,6 +5,7 @@
  *		Ushanka
  *		Pumpkin head
  *		Kitty ears
+ *		Butt
```
important

Probe1 commented:
woop
woop

dats dasound o da police

9600bauds commented:
For the longest time I misheard the lyrics and thought they said "that's the sound that I miss" and it made sense to me because I never heard the rest of the song

Probe1 commented:
been there, brain will remind me of it when I try to sleep.

Hoshi-chan commented:
How is that for lack of a better sound?

You couldn't have picked a better sound.

Blithering commented:
Looks good, how much of the song is played when activated?

Probe1 commented:
about 8 10ths of a second

9600bauds commented:
Literally just the WOOP WOOP

liltiptop commented:
Dank
![security](https://cloud.githubusercontent.com/assets/17549009/14430197/9522ee3a-ffce-11e5-8a15-b0fb19e61a3a.jpg)

Shadowmech88 commented:
You say it plays the sound "when used." Does that mean you have to take the helmet off and click it in your hand in order to play the sound?
I'd prefer if it was a UI button or something like hardsuit helmet lights.

9600bauds commented:
It does have a UI button as well

Blithering commented:
What other helmet types need the attachment? I'm happy to provide sprites for them.

Brolaire commented:
HoS armor patch 😂👌

Blithering commented:
@Brolaire 
![stronger](https://cloud.githubusercontent.com/assets/15067109/14438720/6c322e5e-001f-11e6-9ce2-fec3a242a4bc.png)
HoS with dermal armour: 'I am growing stronger.'

9600bauds commented:
@Blithering If you feel like spriting more hats, I think maybe the Warden's and HoS' hat could use a flashlight attachment. The SWAT helmet is essentially the same sprite as the default helmet, too.

Blithering commented:
@9600bauds  http://s000.tinyupload.com/index.php?file_id=17939740674329619515 There you go, let me know if any others spring to mind.

Brolaire commented:
can the flashlight eyes be a real thing?

Intigracy commented:
Why don't you just make the flashlight an overlay?

9600bauds commented:
I am much too lazy and it wasn't sprited like so

9600bauds commented:
Added @Blithering's FANTASTIC sprites, including the HoS' hat, SWAT helmet, and the Warden's cap. Also added the tactical light as an actual item, it's actually just a flashlight reskin lol
Only one small detail though: If you can sprite something better for the tactical light than this monstrosity I made then it would be appreciated
![dreammaker_2016-04-12_01-42-46](https://cloud.githubusercontent.com/assets/6526157/14450012/40bf8a1a-0050-11e6-9f1b-5d8707229711.png)

9600bauds commented:
Of course the glowing eyes can be a real thing
![mspaint_2016-04-12_01-48-36](https://cloud.githubusercontent.com/assets/6526157/14450051/b79cb6f8-0050-11e6-8999-4c0d2e8a87ec.png)

Shadowmech88 commented:
It'd be neat to be able to attach the tactical light to children of `/obj/item/weapon/gun`.
Maybe make it shine brighter than a flashlight but only in the direction you're facing.

9600bauds commented:
Would require a ton of spritework, and probably wouldn't be too ez to make the directional light actually look good

Shadowmech88 commented:
You could just not make the sprite show up on the gun.

9600bauds commented:
Would be even lazier than my standards, and probably wouldn't be too ez to make the directional light actually look good

9600bauds commented on code/modules/clothing/head/helmet.dm:
```diff
@@ -1,20 +1,37 @@
+/obj/item/clothing/head/helmet/siren/attack_self(mob/user)
+	if(spamcheck)
+		return
+	playsound(get_turf(src), 'sound/voice/woopwoop.ogg', 100, 1, vary = 0)
```
Vary pitch y/n?

Shadowmech88 replied:
I don't think so. Police sirens are supposed to be a constant pitch, right?

9600bauds replied:
I suspect this is essentially going to be a helmet-mounted bike horn that coincidentally happens to WOOP WOOP
there will actually be very little of a siren in the uses people will give it

Intigracy replied:
y

ihadtoregisterforthis replied:
n

Iamgoofball replied:
n, it'll sound weird as fuck

ihadtoregisterforthis commented:
Constant WOOP WOOP is better than everything between "weep weep" and "bwhup bwhup".
Even more if it's a voice clip.

Jeroen52 commented:
Merge when?

9600bauds commented:
when I code it

Brolaire commented:
will this include tacticool eyes or is that too silly?

9600bauds commented:
See above

Blithering commented:
http://s000.tinyupload.com/?file_id=36221188306375178146 Got the on state for the light, but the one you made is fine if you don't want the bother of re-downloading.

9600bauds commented:
ty4spriten

Blithering commented:
No worries, thanks for coding it all.

9600bauds commented:
I just realized we missed the riot helmets
Other than that this is reddy 4 merge

Blithering commented:
http://s000.tinyupload.com/index.php?file_id=08080124128622561331 Riot helmet added, also I realised that the regular helmet's light swapped sides when facing north, so that's fixed.

9600bauds commented:
:clap::clap::clap::clap:

liltiptop commented:
Awesome. But where will we get the tacticool lights? Sec vendors?

9600bauds commented:
The sec helmets spawn with a tacticool light already mounted, and there are one on the Warden's and HoS' locker. They're literally just a flashlight reskin though, you can use regular flashlights too

Brolaire commented:
tfw no flashlight eyes
I might cry now, but i'll get over it

9600bauds commented:
the flashlight eyes are in
this is the third time I tell you that

Shadowmech88 commented:
I like the new additions, but I really wish someone would add the flashlight eyes.

Intigracy commented:
Bauds how come you won't add the flashlight eyes?

You're so mean.

9600bauds commented:
![kcuf](https://cloud.githubusercontent.com/assets/6526157/14568691/d9adc2f6-0310-11e6-98af-8dfef9db350a.jpg)

Jeroen52 commented:
Merge when?

Intigracy commented:
when he adds flashlight eyes :^)

Jeroen52 commented:
Just merge the PR already.

Intigracy commented:
I can't lmao.

Probe1 commented:
![image](https://cloud.githubusercontent.com/assets/8550416/14570548/eec7556a-0312-11e6-85ae-9c17e5637dcf.png)

clusterfack commented on code/modules/clothing/head/helmet.dm:
```diff
@@ -1,51 +1,50 @@
+
+/obj/item/clothing/head/helmet/siren/New()
+	..()
+	set_light(4)
```
So this just has a light 24/7?

PJB3005 replied:
Side note: if you're gonna do that just set `light_range` on the type directly.

9600bauds replied:
Ye, there are no off sprites. Besides, who would ever want to turn it off?

9600bauds replied:
PJB was right with the light_range thing

9600bauds commented:
ty clusterman of optimization

Jeroen52 commented:
IT MERGED!!

Blithering commented:
Its happening.

Intigracy commented on code/modules/clothing/head/helmet.dm:
```diff
@@ -1,51 +1,47 @@
+	if(spamcheck)
+		return
+	playsound(get_turf(src), 'sound/voice/woopwoop.ogg', 100, 1, vary = 0)
+	user.show_message("<span class='warning'>[user]'s [name] rasps, \"WOOP WOOP!\"</span>",1)
```
Should have been visible_message fam.

show_message is the functional equivilant of to_chat

9600bauds replied:
shit

Intigracy replied:
iirc the usage is visible_message("message", get_turf(src))
